examples/named-worker-with-on-fail/named_worker_with_on_fail.bal

```diff
@@ -0,0 +1,28 @@
+import ballerina/io;
+
+public function main() {
+    int[] vals = [2, 3, 4, 5];
+    int key = 1;
+    int val = 0;
+
+    worker w1 {
+        int? index = vals.indexOf(key);
+        if index != () {
+            val = vals[index];
+        } else {
+            check error("value not found");
+        }
```

Comment:
Let's extract if-else logic to a function. Then the usage is very clear. 
```suggestion
    int[] values = [2, 3, 4, 5];
    int value = 1;

    worker w1 {
        int index = check getIndex(values, value);
        index -> w2;
```

where, 
```bal
function getIndex(int[] values, int value) returns int|error {
    int? index = values.indexOf(value);
    if index is () {
        return error("value not found");
    } 
    
    return index;
}
```